tests/e2e/tests/pilot/istio_rbac_test.go

```diff
@@ -56,55 +58,105 @@ func TestRBAC(t *testing.T) {
+		{dst: "a", src: "c", port: 9090, allow: false},
+		{dst: "a", src: "d", path: "/", allow: false},
+		{dst: "a", src: "d", port: 90, allow: false},
+		{dst: "a", src: "d", port: 9090, allow: false},
+
+		{dst: "b", src: "a", path: "/xyz", allow: true},
+		{dst: "b", src: "a", path: "/", allow: true},
+		{dst: "b", src: "a", port: 90, allow: true},
+		{dst: "b", src: "a", port: 9090, allow: true},
+		{dst: "b", src: "c", path: "/", allow: true},
+		{dst: "b", src: "c", port: 90, allow: true},
+		{dst: "b", src: "c", port: 9090, allow: true},
+		{dst: "b", src: "d", path: "/", allow: true},
+		{dst: "b", src: "d", port: 90, allow: true},
+		{dst: "b", src: "d", port: 9090, allow: true},
+
+		{dst: "c", src: "a", path: "/", allow: false},
+		{dst: "c", src: "a", path: "/good", allow: false},
+		{dst: "c", src: "a", path: "/prefixXYZ", allow: false},
+		{dst: "c", src: "a", path: "/xyz/suffix", allow: false},
+		{dst: "c", src: "a", port: 90, allow: false},
+		{dst: "c", src: "a", port: 9090, allow: false},
+
+		{dst: "c", src: "b", path: "/", allow: false},
+		{dst: "c", src: "b", path: "/good", allow: false},
+		{dst: "c", src: "b", path: "/prefixXYZ", allow: false},
+		{dst: "c", src: "b", path: "/xyz/suffix", allow: false},
+		{dst: "c", src: "b", port: 90, allow: false},
+		{dst: "c", src: "b", port: 9090, allow: false},
+
+		{dst: "c", src: "d", path: "/", allow: false},
+		{dst: "c", src: "d", path: "/xyz", allow: false},
+		{dst: "c", src: "d", path: "/good", allow: allow},
+		{dst: "c", src: "d", path: "/prefixXYZ", allow: allow},
+		{dst: "c", src: "d", path: "/xyz/suffix", allow: allow},
+		{dst: "c", src: "d", port: 90, allow: allow},
+		{dst: "c", src: "d", port: 9090, allow: false},
+
+		{dst: "d", src: "a", path: "/xyz", allow: allow},
+		{dst: "d", src: "a", port: 90, allow: false},
+		{dst: "d", src: "a", port: 9090, allow: true},
+		{dst: "d", src: "b", path: "/", allow: allow},
+		{dst: "d", src: "b", port: 90, allow: false},
+		{dst: "d", src: "b", port: 9090, allow: true},
+		{dst: "d", src: "c", path: "/", allow: allow},
+		{dst: "d", src: "c", port: 90, allow: false},
+		{dst: "d", src: "c", port: 9090, allow: true},
 	}
 
 	for _, req := range cases {
 		for cluster := range tc.Kube.Clusters {
-			testName := fmt.Sprintf("%s from %s cluster->%s%s[%s]", req.src, cluster, req.dst, req.path, req.expect)
-			runRetriableTest(t, cluster, testName, defaultRetryBudget, func() error {
-				resp := ClientRequest(cluster, req.src, fmt.Sprintf("http://%s%s", req.dst, req.path), 1, "")
-				if len(resp.Code) > 0 && resp.Code[0] == req.expect {
-					return nil
+			port := ""
+			if req.port != 0 {
+				port = fmt.Sprintf(":%d", req.port)
+			}
+			expectStr := "deny"
+			if req.allow {
+				expectStr = "allow"
+			}
+			testName := fmt.Sprintf("%s from %s cluster->%s%s%s[%s]",
+				req.src, cluster, req.dst, req.path, port, expectStr)
+
+			runRetriableTest(t, cluster, testName, 30, func() error {
```

Comment:
30 => defaultRetryBudget?

Reply:
defaultRetryBudget is 10 but occasionally it takes about 25 seconds in my local environment to finish a single test case, after that each remaining test case finished in less 1 seconds. So I feel we need a larger retry budget.

Reply:
Shall we increase the retry budget?

Reply:
Do you mean the `defaultRetryBudget`? Here I guess 30 should be enough.

Reply:
Yes, I mean defaultRetryBudget.

Reply:
we probably should not to change as existing tests works well with this value.